### PR TITLE
VU macro mode transfer instruction test

### PR DESCRIPTION
**Pull request by jpd002**
This tests the various transfer instructions available in VU macro mode: CFC2, CTC2, QMFC2, QMTC2, LQC2, SQC2.

### CODE REVIEW

File: tests/cpu/vu0_macro/transfer.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+			"addiu $t0, $0, -1\n" \
+			"vnop\n" \
+			"ctc2 $t0, " #reg "\n" \
+			"vnop\n" \
+			"cfc2 %0, " #reg "\n" \
+			"sync\n" \
+			: "=r"(x__printvi) : : "t0" \
+		); \
+		printf("ctc2 -> " #reg ": %08x\n", x__printvi);\
+	} while (false) \
```
Comment (unknownbrackets) [inline tests/cpu/vu0_macro/transfer.cpp]: nit: This line shouldn't have a `\`.

-[Unknown]

File: tests/cpu/vu0_macro/transfer.expected [hunk 2]
```diff
@@ -0,0 +1,35 @@
+-- TEST BEGIN
+lqc2:
+  lqc2 +0: c0dec0de beefdead abcdef01 23456789
+  lqc2 +16: 44556677 00112233 ccddeeff 8899aabb
+  lqc2 -16: c0de1337 deadbeef 9abcdef0 45678123
+  lqc2 -> $vf0: 3f800000 00000000 00000000 00000000
```
Comment (unknownbrackets) [inline tests/cpu/vu0_macro/transfer.expected:6]: Hmm, is this an expected result?

-[Unknown]

**Review by unknownbrackets — COMMENTED** [review]
Looks good, but I'm not sure I understand the "lqc2 -> $vf0" result - is it a predictable result?  Sorry if I'm missing something obvious.

-[Unknown]

File: tests/cpu/vu0_macro/transfer.expected (hunk 2 above)
Comment (jpd002) [reply tests/cpu/vu0_macro/transfer.expected:6]: Yeah, VF0 is hardwired to (0.0f, 0.0f, 0.0f, 1.0f), just checking that LQC2 doesn't change its value. Maybe I should change the value displayed for something with a better meaning?

Comment (unknownbrackets) [reply tests/cpu/vu0_macro/transfer.expected:6]: Oh, duh.  Sorry, that makes sense.

-[Unknown]

**Comment by jpd002** [comment]
Thanks for merging!

**Pull request merged**